src/main/java/com/importer/importer/DataSourceConfig.java

```diff
@@ -19,18 +18,35 @@ public class DataSourceConfig {
 
     @Value("${spring.datasource.username}")
     private String dataBaseUsername;
+    
+    @Value("${spring.datasource.password}")
+    private String dataBasePassword;
 
     @Autowired
-    private Environment environment;
+    private KMSUtil kmsUtil;
 
     @Bean
     public DataSource dataSource() throws IOException {
-        String dbPassword = environment.getProperty("spring.datasource.password");
+       // String dbPassword = environment.getProperty("spring.datasource.password");
+        boolean isEncrypted = checkDbPasswordIsEncrypted(dataBasePassword);
+        if(isEncrypted){
+          dataBasePassword = decryptEncryptedPassword(dataBasePassword);
+        }
         DriverManagerDataSource dataSource = new DriverManagerDataSource();
         dataSource.setDriverClassName("org.postgresql.Driver");
         dataSource.setUrl(dataBaseUrl);
         dataSource.setUsername(dataBaseUsername);
-        dataSource.setPassword(dbPassword);
+        dataSource.setPassword(dataBasePassword);
         return dataSource;
     }
+
+    private boolean checkDbPasswordIsEncrypted(String dataBasePassword) {
```

Comment:
Do we need 2 methods- checkDbPasswordIsEncrypted and decryptEncryptedPassword? 
Can't we club both methods into one?

If prop is enclosed in enc then decrypt else return the value.